nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -523,4 +528,49 @@ public void close() {
       api.close();
     }
   }
+
+  public TableMetadata loadTableMetadataWithNessieSpecificProperties(
+      String metadataLocation, IcebergTable table, FileIO fileIO, String identifier) {
+    // Update the TableMetadata with the Content of NessieTableState.
+    TableMetadata deserialized = TableMetadataParser.read(fileIO, metadataLocation);
+    Map<String, String> newProperties = Maps.newHashMap(deserialized.properties());
+    newProperties.put(NessieTableOperations.NESSIE_COMMIT_ID_PROPERTY, getRef().getHash());
```

Comment:
Consistency between `metadataLocation` and `getRef()` is essentially ensured at the caller level. I think it makes sense to keep both of them as parameters to this method to emphasize that it is the responsibility of the caller to provide matching values for them.

Reply:
If you're ok with this point and the comment about `FileIO`, then perhaps this method could be in `NessieUtil` as it will have little to do with the Nessie "Client" per se. WDYT?

Reply:
> then perhaps this method could be in NessieUtil as it will have little to do with the Nessie "Client" per se. WDYT?

If I move this as a static method to NessieUtil, the Trino caller cannot pass the `Reference` as `NessieIcebergClient.getRef` returns the `UpdateableReference` which is a package private. So, it forces me to change `UpdateableReference` as public class.

Reply:
`NessieIcebergClient.getRef()` actually looks like a rather internal method. Perhaps it should be package-private. We could add a new public `getReference()` method to return the current Nessie API `Reference`. WDYT?

Reply:
- added the `getReference` public method and made `getRef` as package private.
- Moved the whole method to `NessieUtil`